### PR TITLE
Feat/logger_map

### LINKED ISSUE #10636: [Trainer] dictionary access to multiple named loggers
## 🚀 Feature

When using multiple loggers, being able to give a dict of loggers to the `Trainer()` instead of a list.

### Motivation

I find the current list access when calling a single logger not explicit.

### Pitch

```python
trainer = Trainer(
    logger={
        'tensorboard': TensorBoardLogger(...),
        'csv': CSVLogger(...)},
    ...
)
```
then in the loop:
```python
self.logger['tensorboard'].experiment.add_image()
```
instead of
```python
self.logger[0].experiment.add_image()
```

There could be an ambiguity when using an integer as key, so maybe enforce string-only keys.

### PR DESCRIPTION
**Pull request by GdoongMathew**
## What does this PR do?

Fixes #10636

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21297.org.readthedocs.build/en/21297/

### CODE REVIEW

**Review by SkafteNicki — CHANGES_REQUESTED** [review]
Going to block until we have agreed on a interface in https://github.com/Lightning-AI/pytorch-lightning/issues/10636
